### PR TITLE
SD-3054 - add Http Push type of delivery to be used by Content API subscriber

### PR DESCRIPTION
**Pull request by mugurrus**

### CODE REVIEW

File: client/app/scripts/superdesk-publish/views/http-push-config.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div class="field">
+    <label for="ftp_host" translate>Resource URL</label>
```
Comment (amagdas) [inline client/app/scripts/superdesk-publish/views/http-push-config.html]: 'http_host' ?

File: client/app/scripts/superdesk-publish/views/http-push-config.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<div class="field">
+    <label for="http_host" translate>Resource URL</label>
+    <input type="text" id="http_host" placeholder="{{ :: 'Resource URL'|translate }}" ng-model="destination.config.resource_url" required ng-change="$parent.setConfig(destination)">
+</div>
+<div class="field">
+    <label for="http_host" translate>Item URL</label>
+    <input type="text" id="http_host" placeholder="{{ :: 'Item URL'|translate }}" ng-model="destination.config.item_url" required ng-change="$parent.setConfig(destination)">
```
Comment (plamut) [inline client/app/scripts/superdesk-publish/views/http-push-config.html]: Would it make sense to defer the model update by a few hundred milliseconds, i.e. when the user has likely finished with typing?

**Comment by ioanpocol** [comment]
It will help to have an e2e and/or behave test that proves that the http push delivery is working well.

**Comment by plamut** [comment]
I agree with @ioanpocol on tests, they are needed.

**Comment by mugurrus** [comment]
An e2e test for http push doesn't make much sense. HTTP push is run in a celery task and pushes content to an external API. The chain of involved components is long and there's not much info you get if the crash happens in the celery task. I think it's better to test for the item insertion in the publish queue.

**Comment by petrjasek** [comment]
:+1:

**Comment by amagdas** [comment]
:+1:

**Pull request merged**